### PR TITLE
Supporting generic iCal feeds

### LINKED ISSUE #1985: Calendar URL field is giving a false negative "not supported" message if using a generic ical feed
Currently the default is to auto-detect the importer mode that needs to be used when adding a calendar, however this doesn't work for some URLs, especially self-hosted .ics files. The text description should prompt the uploader to check the importer mode is appropriate, especially if their URL fails to import the first time. 

## Acceptance criteria

- [x] `https://www.birchcommunitycentre.co.uk/events.ics` can be added as a calendar url without getting told it cant be parsed
- [x] `https://membership.ambition.rip/public_calendar.ics` can be added as a calendar url without getting told it cant be parsed
- [ ] URLs we don't recognise can still be saved in general so the admin team can have a look at them

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Closes #1985 

If the URL starts with `https://` and ends in `.ics` then it is an ical feed.

### CODE REVIEW

**Comment by kimadactyl** [comment]
also `webcal://`!

**Comment by ivan-kocienski-gfsc** [comment]
> also `webcal://`!
Still supported by the earlier patterns.

> 
> also plain http?
fine.

File: test/jobs/calendar_importer/calendar_importer_task_test.rb [hunk 1]
```diff
@@ -189,4 +189,26 @@ class CalendarImporterTaskTest < ActiveSupport::TestCase
       assert_equal "Source responded with invalid JSON (783: unexpected token at '{ \"key\": \"va')", error.message
     end
   end
+
+  test 'can import from generic iCal feed' do
+    VCR.use_cassette(:generic_ical_feed, allow_playback_repeats: true) do
+      calendar = create(
+        :calendar,
+        name: 'Generic iCal Calendar',
+        source: 'https://www.birchcommunitycentre.co.uk/events.ics',
```
Comment (aaaaargZombies) [inline test/jobs/calendar_importer/calendar_importer_task_test.rb:198]: can we get tests for the `http` and `webcal` type links that got added (I'm assuming) after this test was written.

**Review by aaaaargZombies — CHANGES_REQUESTED** [review]
small change request then good to go, I think Kim raises some good points about having very specific URLs for partners in there but they are outside the scope of this issue and shouldn't hold up this important feature.

File: test/jobs/calendar_importer/calendar_importer_task_test.rb (hunk 1 above)
Comment (ivan-kocienski-gfsc) [reply test/jobs/calendar_importer/calendar_importer_task_test.rb:198]: I put in a test for webcal. i can put in *a* test for `http` but to be thorough and check all the patterns i'd have to write an `http` variant for every type of ical feed we support.

Comment (ivan-kocienski-gfsc) [reply test/jobs/calendar_importer/calendar_importer_task_test.rb:198]: if that is something that would be valuable then let me know

**Review by aaaaargZombies — APPROVED** [review]
cheers, looks good, wild we didn't have this sooner

**Comment by kimadactyl** [comment]
> We also should more gracefully handle an unknown URL and make a suggestion to try one of the generic settings.

I don't think this was addressed in the PR

**Pull request merged**